python/lsst/ts/MTAOS/model.py

```diff
@@ -352,14 +433,182 @@ async def run_wep(self, visit_id, extra_id, config, **kwargs):
+        )
+
+        comcam_config_file = tempfile.NamedTemporaryFile(suffix=".yaml")
+
+        comcam_config_file.write(yaml.safe_dump(wep_configuration).encode())
+
+        comcam_config_file.flush()
+
+        self.log.debug(f"Pipeline configuration in {comcam_config_file.name}.")
+
+        run_pipetask_cmd = writePipetaskCmd(
+            self.data_path,
+            self.run_name,
+            self.pipeline_instrument["comcam"],
+            "refcats," + self.collections,
+            pipelineYaml=comcam_config_file.name,
+        )
 
-        # TODO: (DM-28710) Initial implementation of runWEP command in MTAOS
-        raise NotImplementedError("This function is not supported yet (DM-28710).")
+        run_pipetask_cmd += f' -d "exposure IN ({intra_id}, {extra_id})"'
+        run_pipetask_cmd += f" -j {self.pipeline_n_processes}"
+
+        self.log.debug(f"Running: {run_pipetask_cmd}")
+
+        # Run pipeline task in a process asynchronously
+        wep_process = await asyncio.create_subprocess_shell(
+            run_pipetask_cmd,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+        )
+
+        stdout, stderr = await wep_process.communicate()
+
+        self.log.debug(f"Process returned: {wep_process.returncode}")
+
+        self.log.debug(stdout)
+
+        self.log.error(stderr)
+
+        self.log.debug("Data processing completed. Gathering output.")
+
+        comcam_config_file.close()
+
+        butler = dafButler.Butler(self.data_path)
+
+        # We may need to run the following in an executor so we won't block the
+        # event loop.
+
+        datasetRefs = list(
+            butler.registry.queryDatasets(
+                datasetType="postISRCCD", collections=[self.run_name]
+            )
+        )
+        for ref in datasetRefs:
+            self.log.debug(ref.dataId)
+
+        # Get output
+        data_ids = butler.registry.queryDatasets(
+            self.zernike_table_name,
+            dataId=dict(
+                instrument=self.data_instrument_name["comcam"], exposure=extra_id
+            ),
+            collections=[self.run_name],
+        )
+
+        self.log.debug(f"Processing yielded: {data_ids}")
+
+        self.wavefront_errors.append(
+            [
+                (
+                    data_id.dataId["detector"],
+                    butler.get(
+                        self.zernike_table_name,
+                        dataId=data_id.dataId,
+                        collections=[self.run_name],
+                    ),
+                )
+                for data_id in data_ids
+            ]
+        )
+
+    def generate_wep_configuration(self, instrument, reference_id, config):
+        """Generate configuration dictionary for running the WEP pipeline based
+        on a reference image id and a configuration dictionary.
+
+        Parameters
+        ----------
+        instrument: `str`
```

Comment:
instrument: `str` --> instrument : `str`

Reply:
if we use "`" documenteer creates links to the types: e.g.  https://ts-mtaos.lsst.io/py-api/lsst.ts.MTAOS.Model.html#lsst.ts.MTAOS.Model.add_correction